assets/wizards/popups/index.js

```diff
@@ -0,0 +1,251 @@
+			method: 'POST',
+			data: {
+				categories,
+			},
+		} )
+			.then( ( { popups } ) => this.setState( { popups: this.sortPopups( popups ) } ) )
+			.catch( error => setError( error ) );
+	};
+
+	updatePopup = ( popupId, options ) => {
+		const { setError, wizardApiFetch } = this.props;
+		console.log( options );
+		return wizardApiFetch( {
+			path: `/newspack/v1/wizard/newspack-popups-wizard/${ popupId }`,
+			method: 'POST',
+			data: { options },
+		} )
+			.then( ( { popups } ) => this.setState( { popups: this.sortPopups( popups ) } ) )
+			.catch( error => setError( error ) );
+	};
+
+	/**
+	 * Delete a popup.
+	 *
+	 * @param {number} popupId ID of the Popup to alter.
+	 */
+	deletePopup = popupId => {
+		const { setError, wizardApiFetch } = this.props;
+		return wizardApiFetch( {
+			path: `/newspack/v1/wizard/newspack-popups-wizard/${ popupId }`,
+			method: 'DELETE',
+		} )
+			.then( ( { popups } ) => this.setState( { popups: this.sortPopups( popups ) } ) )
+			.catch( error => setError( error ) );
+	};
+
+	/**
+	 * Sort Pop-ups into categories.
+	 */
+	sortPopups = popups => {
+		const overlay = this.sortPopupGroup(
+			popups.filter( ( { options } ) => 'inline' !== options.placement )
+		);
+		const inline = this.sortPopupGroup(
+			popups.filter( ( { options } ) => 'inline' === options.placement )
+		);
+		return { overlay, inline };
+	};
+
+	/**
+	 * Sort Pop-up groups into categories.
+	 */
+	sortPopupGroup = popups => {
+		const test = popups.filter( ( { options } ) => 'test' === options.frequency );
+		const active = popups.filter( ( { categories, options, sitewide_default: sitewideDefault } ) =>
+			'inline' === options.placement
+				? 'test' !== options.frequency && 'never' !== options.frequency
+				: 'test' !== options.frequency && ( sitewideDefault || categories.length )
+		);
+		const activeWithSitewideDefaultFirst = [
+			...active.filter( ( { sitewide_default: sitewideDefault } ) => sitewideDefault ),
+			...active.filter( ( { sitewide_default: sitewideDefault } ) => ! sitewideDefault ),
+		];
+		const inactive = popups.filter(
+			( { categories, options, sitewide_default: sitewideDefault } ) =>
+				'inline' === options.placement
+					? 'never' === options.frequency
+					: 'test' !== options.frequency && ( ! sitewideDefault && ! categories.length )
+		);
+		return { test, active: activeWithSitewideDefaultFirst, inactive };
+	};
+
+	previewUrlForPopup = ( { options, id } ) => {
+		const { placement, trigger_type: triggerType } = options;
+		const previewURL =
+			'inline' === placement || 'scroll' === triggerType
+				? window && window.newspack_popups_data && window.newspack_popups_data.preview_post
+				: '/';
+		return `${ previewURL }?newspack_popups_preview_id=${ id }`;
+	};
+
+	render() {
+		const { pluginRequirements } = this.props;
+		const { popups, previewUrl } = this.state;
+		const { inline, overlay } = popups;
+		return (
+			<Fragment>
```

Comment:
Nitpick: `Fragment`'s not needed here since it wraps a single component

Reply:
https://github.com/Automattic/newspack-plugin/pull/515/commits/a0b4e70d44ea62e3db646667c9cfbd6c972261c6